examples/lib/defaultHtmlSequencerUi.js

```diff
@@ -29,7 +29,7 @@ function DefaultHtmlSequencerUi(_sequencer, options) {
 
   function removeStepUi() {
     var index = $(removeStepSel).index(this) + 1;
-    sequencer.removeSteps(index).run();
+    sequencer.removeSteps(index).run(sequencer.images.image1.steps.length-1);
```

Comment:
Here's where maybe we should think about whether `run` should auto-detect that nothing has changed in certain steps? Or do you think it makes sense to do this in the UI only?

Reply:
@jywarren I think this only makes sense in the UI !!

Reply:
Is that because when we are in Node, we aren't immediately running `run()` when we add steps? I almost think that would be a scenario where we'd want the sequencer to know that nothing has changed on prior steps, automatically, and not have to re-run the math for prior steps. 

Like each step would potentially have a `.changed` property, and doing `.insertStep()` would switch that to `true` for any subsequent step. What do you think of this "switch-based" idea?

Reply:
@jywarren I don't fully understand sorry, in node we run one predefined sequence right? So how would we be able to call insertStep in node?? I mean we can't modify the sequence during execution in node right?

Reply:
Well, i think we can set this aside for now based on my other comments. But just imagine -- 

```js
var sequencer = require("../image-sequencer")();
...
sequencer.run();
sequencer.insertSteps('blur', 2);
sequencer.run(); // here, we wouldn't have to re-run steps 1 or 2, right?
```

Reply:
@jywarren Oh I got it!!! I was confuding node.js as CLI!! Obviously we need this functionality in node yeah!! let's do the thing where we pass in the values as keys in an object like `{index: ,progressObj: }`